crates/apub/src/lib.rs

```diff
@@ -34,13 +35,39 @@ static CONTEXT: Lazy<Vec<serde_json::Value>> = Lazy::new(|| {
 fn local_instance(context: &LemmyContext) -> &'static LocalInstance {
   static LOCAL_INSTANCE: OnceCell<LocalInstance> = OnceCell::new();
   LOCAL_INSTANCE.get_or_init(|| {
+    let conn = &mut context
+      .pool()
+      .get()
+      .expect("getting connection for LOCAL_INSTANCE init");
+    // Local site may be missing
+    let local_site = &LocalSite::read(conn);
+    let worker_count = local_site
+      .as_ref()
+      .map(|l| l.federation_worker_count)
+      .unwrap_or(64) as u64;
+    let http_fetch_retry_limit = local_site
+      .as_ref()
+      .map(|l| l.federation_http_fetch_retry_limit)
+      .unwrap_or(25);
+    let federation_debug = local_site
+      .as_ref()
+      .map(|l| l.federation_debug)
+      .unwrap_or(true);
+
     let settings = InstanceSettings::builder()
-      .http_fetch_retry_limit(context.settings().federation.http_fetch_retry_limit)
-      .worker_count(context.settings().federation.worker_count)
-      .debug(context.settings().federation.debug)
+      .http_fetch_retry_limit(http_fetch_retry_limit)
+      .worker_count(worker_count)
+      .debug(federation_debug)
       // TODO No idea why, but you can't pass context.settings() to the verify_url_function closure
       // without the value getting captured.
-      .verify_url_function(|url| check_apub_id_valid(url, &SETTINGS))
+      // TODO this is broken
+      // .verify_url_function(|url| {
+      //   let conn2 = &mut context.to_owned().pool().get().expect("getting connection for LOCAL_INSTANCE init");
+      //   let local_site = &LocalSite::read(conn2).expect("reading local_site");
+      //   let allowed_instances = &Instance::allowlist(conn2).ok();
+      //   let blocked_instances = &Instance::blocklist(conn2).ok();
+      //   check_apub_id_valid(url, local_site, allowed_instances, blocked_instances, &SETTINGS)
+      // })
```

Comment:
Dont forget to fix this.

Reply:
I spent like an hour trying to fix this one and failed, I need your help with it.

Seems like the only way to fix is to have your verify_url_function either accept a closure rather than a function.